packages/lib/models/Note.ts

```diff
@@ -830,7 +830,7 @@ export default class Note extends BaseItem {
 		if (dispatchUpdateAction) {
 			// Ensures that any note added to the state has all the required
 			// properties for the UI to work.
-			if (!('deleted_time' in savedNote)) {
+			if (!('deleted_time' in savedNote) || !('share_id' in savedNote)) {
```

Comment:
This seems to be causing a number of tests to fail (perhaps because it causes additional `Note.load`s when `share_id` isn't specified?). An alternate fix might be to adjust [itemIsReadOnlySync](https://github.com/laurent22/joplin/blob/5a44f62fb6988a5b3380cc86274fd20947952c73/packages/lib/models/utils/readOnly.ts#L89) to not `throw` if a note is missing `share_id`.

Reply:
It's throwing because otherwise it would be difficult to know that something is wrong somewhere, and invalid data would get into the state, so it would be better to keep this code

Reply:
I've updated the diff such that `share_id` does not cause the note to be loaded from the database (which was causing tests to fail).

I'm converting this pull request to a draft until I've re-done the above manual testing steps.